packages/ember-application/tests/system/application_test.js

```diff
@@ -24,389 +24,404 @@ import {
+      this.createApplication();
+      this.addTemplate('application', '{{outlet}}');
+      this.addTemplate('index', '<h1>Hi from index</h1>');
     });
 
-    app.Router.reopen({
-      location: 'none'
-    });
+    assert.equal(jQuery('#qunit-fixture h1').text(), 'Hi from index');
+  }
 
-    app.register('template:application', compile('<h1>Hello!</h1>'));
-  });
+  [`@test ready hook is called before routing begins`](assert) {
+    assert.expect(2);
 
-  let router = app.__container__.lookup('router:main');
-  equal(router instanceof Router, true, 'Router was set from initialize call');
-  equal(jQuery('#qunit-fixture h1').text(), 'Hello!');
-});
+    run(() => {
+      function registerRoute(application, name, callback) {
+        let route = EmberRoute.extend({
+          activate: callback
+        });
 
-QUnit.test('ApplicationView is inserted into the page', function() {
-  run(() => {
-    app = Application.create({
-      rootElement: '#qunit-fixture'
-    });
+        application.register('route:' + name, route);
+      }
 
-    setTemplate('application', compile('<h1>Hello!</h1>'));
+      let MyApplication = Application.extend({
+        ready() {
+          registerRoute(this, 'index', () => {
+            assert.ok(true, 'last-minute route is activated');
+          });
+        }
+      });
 
-    app.ApplicationController = Controller.extend();
+      let app = this.createApplication({}, MyApplication);
 
-    app.Router.reopen({
-      location: 'none'
+      registerRoute(app, 'application', () => ok(true, 'normal route is activated'));
     });
-  });
-
-  equal(jQuery('#qunit-fixture h1').text(), 'Hello!');
-});
+  }
 
-QUnit.test('Minimal Application initialized with just an application template', function() {
-  jQuery('#qunit-fixture').html('<script type="text/x-handlebars">Hello World</script>');
-  app = run(() => {
-    return Application.create({
-      rootElement: '#qunit-fixture'
+  [`@test initialize application via initialize call`](assert) {
+    run(() => {
+      this.createApplication();
     });
-  });
+    // This is not a public way to access the container; we just
+    // need to make some assertions about the created router
+    let router = this.application.__container__.lookup('router:main');
+    assert.equal(router instanceof Router, true, 'Router was set from initialize call');
+    assert.equal(router.location instanceof NoneLocation, true, 'Location was set from location implementation name');
+  }
 
-  equal(trim(jQuery('#qunit-fixture').text()), 'Hello World');
-});
+  [`@test initialize application with stateManager via initialize call from Router class`](assert) {
+    run(() => {
+      this.createApplication();
+      this.addTemplate('application', '<h1>Hello!</h1>');
+    });
+    // This is not a public way to access the container; we just
+    // need to make some assertions about the created router
+    let router = this.application.__container__.lookup('router:main');
```

Comment:
as mentioned above, we should use `__deprecatedInstance__` instead, accessing `application.__container__` should have been deprecated long ago but was apparently not...